martypy/ClientMV2.py

```diff
@@ -407,6 +408,95 @@ def is_conn_ready(self) -> bool:
             return False
         return self._initComplete and (self.lastSubscribedMsgTime is not None)
 
+    def _valid_addon(self, add_on: str) -> bool:
+        disco = {"00000087","00000088","00000089"}
+        for attached_add_on in self.get_add_ons_status().values():
+            if attached_add_on['name'] == add_on:
+                if attached_add_on['whoAmITypeCode'] in disco:
+                    return True
+                else:
+                    raise MartyCommandException("The add on name passed in is not a valid disco add on. Please check the add on name in the scratch app -> configure -> add ons")
+        raise MartyCommandException("The add on name passed in is not a valid add on. Please check the add on name in the scratch app -> configure -> add ons")
+
+    def _valid_hex(self, color_hex: str) -> bool:
+        hex_pattern = re.compile("^([A-Fa-f0-9]{6}|[A-Fa-f0-9]{3})$")
+        return bool(hex_pattern.match(color_hex))
+
+    def disco_off(self, add_on: str) -> bool:
+        if self._valid_addon(add_on):
+            return self.add_on_query(add_on, bytes.fromhex('01'), 0)
+
+    def disco_pattern(self, pattern: str, add_on: str) -> bool:
```

Comment:
Shouldn't pattern be an `int`?
```suggestion
    def disco_pattern(self, pattern: int, add_on: str) -> bool:
```
(If so, a similar change is needed in the other `Client` classes.)